connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Connect.java

```diff
@@ -47,6 +50,11 @@ public Connect(H herder, ConnectRestServer rest) {
         shutdownHook = new ShutdownHook();
     }
 
+    // public for testing
+    public Future<?> getDistributedHerderFuture() {
```

Comment:
Nit: We don't use the `get` prefix in this codebase. Also, this method can technically be called for more than just distributed workers. Maybe we could call it `herderTask()` and note in the Javadocs that it will return null if the herder type doesn't have a separate work thread?